docs/index.rst

```diff
@@ -27,123 +27,316 @@ EdgeDB TypeScript/JS Client
+  without a file extension throughout this documentation. Please see your build
+  or environment tooling's guidance on how to adapt this style.
+
+Query
+^^^^^
+
+Now, let's write a Node.js script that queries the database for details about
+Iron Man 2:
+
+.. code-block:: typescript
+  :caption: query.ts
+
+  import * as edgedb from "edgedb";
+
+  const client = edgedb.createClient();
+
+  async function main() {
+    const result = await client.querySingle(`
+      select Movie {
+        title,
+        actors: {
+          name,
+        }
+      } filter .title = "Iron Man 2"
+    `);
+
+    console.log(JSON.stringify(result, null, 2));
+  }
+
+  main();
+
+Interfaces
+^^^^^^^^^^
+
+Since we're using TypeScript, it would be nice to be able to type the return
+value of this query, so let's use our first generator, the :ref:`Queries
+generator <edgedb-js-queries>` to tell TypeScript what the type of our result
+is.
+
+First we run the generator:
+
+.. code-block:: bash
+
+  $ npx @edgedb/generate interfaces
 
-  const edgedb = require("edgedb");
+This generator introspects your database schema and generates a set of
+equivalent TypeScript interfaces.
+
+Now we can annotate our query since we are selecting the whole ``Movie`` type:
+
+.. code-block:: typescript
+  :caption: query.ts
+
+  import * as edgedb from "edgedb";
+  import { Movie } from "./dbschema/interfaces"
 
   const client = edgedb.createClient();
-  const query = `select "Hello world!";`;
 
-  async function run(){
-    const result = await client.query(query)
-    console.log(result); // "Hello world!"
+  async function main() {
+    // result will be inferred as Movie | null
+    const result = await client.querySingle<Movie>(`
+      select Movie {
+        title,
+        actors: {
+          name,
+        }
+      } filter .title = "Iron Man 2"
+    `);
+
+    console.log(JSON.stringify(result, null, 2));
   }
 
-  run();
+  main();
 
-If you're not using TypeScript, you can skip straight to :ref:`the Client docs
-<edgedb-js-driver>`.
+You can now run the script with ``tsx``:
 
+.. code-block:: bash
 
-.. _edgedb-js-intro-qb:
+  $ npx tsx query.ts
 
-The query builder
-=================
+Queries generator
+^^^^^^^^^^^^^^^^^
+
+Wouldn't it be great if we could write any arbitrary query and get a type-safe
+function that we could call? Good news, that's exactly what the next generator
+does! The :ref:`queries generator <edgedb-js-queries>` Scans your project for
+``*.edgeql`` files and generates a file containing a strongly-typed function.
+
+First, move the query into a separate file called ``getMovie.edgeql``. Next,
+we'll run the ``queries`` generator:
+
+.. code-block:: bash
 
-The EdgeDB query builder provides a **code-first** way to write
-**fully-typed** EdgeQL queries with TypeScript. We recommend it for TypeScript
-users and JavaScript users who prefer writing queries as code.
+  $ npx @edgedb/generate queries --file
```

Comment:
The current implementation is actually broken (see #622 ), but furthermore this exports all potential queries to a centralized file such that you import all of the generated queries from the same file, similar to the interfaces and query builder generators.

I don't think it's worth really talking through this _too_ much since I [plan on changing the default to be the current `--file` behavior](https://github.com/edgedb/edgedb-js/issues/540#issuecomment-1545952603) in the very near future, so this will just go away soon.